pandasai/__init__.py

```diff
@@ -235,7 +235,7 @@ def conversational_answer(self, question: str, answer: str) -> str:
 
     def run(
         self,
-        data_frame: pd.DataFrame,
+        data_frame: Union[pd.DataFrame, List[pd.DataFrame]],
```

Comment:
How do we feel about using the pipe operator instead of Union?
```python
from __future__ import annotations
...
data_frame: pd.DataFrame | list[pd.DataFrame]
```

Reply:
@jonbiemond IIRC `|` is only avaialble starting from python 3.10, so for the time being we should maybe use Union until we drop the support for older versions! What do you think?

Reply:
> How do we feel about using the pipe operator instead of Union?
> 
> ```python
> from __future__ import annotations
> ...
> data_frame: pd.DataFrame | list[pd.DataFrame]
> ```

I had the same thought at first, but as @gventuri has noted, it'd rather be better to use `Uniton` until we drop supports for <3.10 versions.

Reply:
That's where `from __future__ import annotations` comes in. 😀 It lets us use those `typing` features from `3.10` in `3.7` onwards! But really a minor thing and I'm sure there's a few places in the code we can replace imports of the `typing` module. Could be a little project for me one day.